### ISSUE TITLE
Start a release doesnt include actually starting a release

### ISSUE THREAD

**Issue by StingyJack** [MicrosoftDocs/vsts-rest-api-specs#170]
After the release is created, it needs to be deployed to a stage or stages. How is that accomplished using the REST API (or other non clicky means)?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e71b7f22-5cdb-ef2f-ccb1-66cd4835a858
* Version Independent ID: 06106c51-b396-9d97-9566-72e7a27375eb
* Content: [Release REST API for Azure DevOps Services - Azure DevOps Services REST API](https://docs.microsoft.com/en-us/rest/api/azure/devops/release/?view=azure-devops-rest-5.0)
* Content Source: [docs-ref-conceptual/5.0/release/index.md](https://github.com/MicrosoftDocs/vsts-docs-rest-apis/blob/live/docs-ref-conceptual/5.0/release/index.md)
* Service: **vsts**
* Product: **devops**
* GitHub Login: @wnjenkin
* Microsoft Alias: **elbatk**

**Comment by justcallmetrell** [comment]
I am also interested in this operation. How are the Release Pipelines invoked/deployed once created?

**Comment by vithati** [comment]
@StingyJack @justcallmetrell , Please refer [REST API 5.1 version 'Releases - Update Release Environment'](https://docs.microsoft.com/en-us/rest/api/azure/devops/release/releases/update%20release%20environment?view=azure-devops-rest-5.1) for example.

**Comment by StingyJack** [comment]
@vithati - thanks but I'm not looking for how to update a release environment.

**Comment by vithati** [comment]
@StingyJack , [Update release environment](https://docs.microsoft.com/en-us/rest/api/azure/devops/release/releases/update%20release%20environment?view=azure-devops-rest-5.1) rest api used to start stage, schedule stage or cancel a stage. See [this](https://docs.microsoft.com/en-us/rest/api/azure/devops/release/releases/update%20release%20environment?view=azure-devops-rest-5.1#examples) example to start deployment.

**Comment by pabrams** [comment]
The issue described in the title does not match the issue described in the description.  What is the issue?  

If the issue is "Start a release doesn't include actually starting a release", then the solution is probably to rename the "Start a release" heading on the [linked page](https://docs.microsoft.com/en-us/rest/api/azure/devops/release/?view=azure-devops-rest-5.0) to "Create a release", since that heading would more accurately reflect the steps contained thereunder.  

If the issue is the question "How can I deploy a release to a stage or stages using the REST API", then I think you should go to StackOverflow, or do some more research, because the page you linked to is not where you would find that.

**Comment by StingyJack** [comment]
@pabrams - i was looking for how to queue a release in the same vein as queueing a build. 

The titles of these things are all so "off",  your rename suggestion is spot on.  @vithati 's suggestion to use "update release environment" is just as confusing because I dont want to manage data about a release environment entity (not even sure what that is in the context of a release pipeline with stages). I wanted to start a release into the pipeline.

**Comment by pabrams** [comment]
@StingyJack If I'm not mistaken, "Stage" is just a re-label of "Environment", and it specifies where you want to deploy to.   The linked page describes an API call that has parameters such as [scheduledDeploymentTime](https://docs.microsoft.com/en-us/rest/api/azure/devops/release/releases/update%20release%20environment?view=azure-devops-rest-5.1#request-body), which seems like what you want, although I don't see anything to simply queue a release immediately (you could set the time to now, though).  Maybe this [Release Resource](https://docs.microsoft.com/en-us/rest/api/azure/devops/release/releases/update%20release%20resource?view=azure-devops-rest-5.1) stuff is what you want.  
Also, I usually have my releases set up to queue themselves automatically when a build completes, so that might be another option, and then you can just queue the build. 
I think you should submit a separate bug if there's no mention anywhere in the docs of stage==environment.

**Comment by StingyJack** [comment]
Since I found the manual intervention task, and that it can be used for a proper "wait" state (code deployed for testing, but that hasnt been tested yet, isnt sending out a "approve production release" approval), I dont have to code something and thus dont need this api really. 

I do want the docs team on this set of programs to address the many mismatches and gaps like this, but 
an issue opened for "Be consistent between documentation and program terminology" would be asked for a specific instance to address and nullify the original request that has a larger scope than a single page. Hopefully @vithati reads through this part of the conversation and understands this was the idea I was originally trying to convey by example.

**Comment by pabrams** [comment]
Yeah, you make excellent points.  I've opened several ms-docs-related issues in the past couple of years, and almost every time, the issue is just one specific example of a problem that is endemic across all docs (well, perhaps within a particular domain, e.g. TFS web extensions) and all pages.  The people they have tending the docs repos' issue trackers don't have the cognitive ability to generalize; they can only address very specific issues, one at a time.
